lib/ably/modules/channels_collection.rb

```diff
@@ -13,14 +13,17 @@ def initialize(client, channel_klass)
     # Return a Channel for the given name
     #
     # @param name [String] The name of the channel
-    # @param channel_options [Hash] Channel options including the encryption options
+    # @param channel_options [Hash, Ably::Models::ChannelOptions] A hash of options or a {Ably::Models::ChannelOptions}
     #
     # @return [Channel]
     #
     def get(name, channel_options = {})
       if channels.has_key?(name)
         channels[name].tap do |channel|
-          channel.update_options channel_options if channel_options && !channel_options.empty?
+          if channel_options && channel_options.size.nonzero?
+            channel.raise_on_reattach! if channel.respond_to?(:raise_on_reattach!)
+            channel.options = channel_options
```

Comment:
> channel.raise_on_reattach! suggests that it can raise an error (bang method) but it does not because raise_on_reattach! marks channel to be attached.

I believe it's okay to use bang methods for changing state. For example, all ruby methods with bang change state and not raise error

> In your code it will raise an error on setOptions (or set_options/update_options in Ruby).
So here is the spec for raising an error
https://github.com/ably/ably-ruby/pull/336/files#diff-592403190fbe877c053dcd4a89f489fe0b6ab8e8a1c8bf147592fc83dbbbb89cR35

I believe it's not very important where we raise it since it's artificial API limitation there is no point in stacktrace here. Error clearly says you should set_options directly

Reply:
Spec says it should raise an error "if a new set of ChannelOptions is supplied":

```
(RTS3c1) If a new set of ChannelOptions is supplied to Channels#get that would trigger a reattachment of the channel if supplied to Channel#setOptions per RTL16a, it must raise an error, informing the user that they must use Channel#setOptions instead
```

I have made a build and tested this case,
and it does not work. Commented here also https://github.com/ably/ably-ruby/pull/336#issuecomment-1035444380

Reply:
@TheSmartnik will deal with this one